src/ch/ch_monitor.c

```diff
@@ -255,6 +256,18 @@ virCHMonitorBuildNetJson(virJSONValuePtr nets, virDomainNetDefPtr netdef)
             if (virJSONValueObjectAppendString(net, "mask", netmaskStr) < 0)
                 goto cleanup;
         }
+
+        /* network and bridge use a tap device, and direct uses a
+         * macvtap device
+         */
+        if (/*driver->privileged && */nicindexes && nnicindexes &&
```

Comment:
Does this break when the driver isn't running as root?

Reply:
Composing json that is passed to cloud-hypervisor socket should not require root privileges. Seems like that is why it is commented out here. I will clean up the comments.

Reply:
Cleaned up.